_keystone/saml-federation.md

```diff
@@ -0,0 +1,541 @@
+---
+layout: page
+title: Keystone SAML Federation on IBM Blue Box Cloud
+featured: false
+tags: [federation, keystone, saml]
+dateAdded: September 26, 2016
+author: Elvin Tubillara
+editor: ##WHO WHO WHO?
+---
+
+* [What is SAML Federation?](#what_is_saml?)
+* [Integrating the IDP and Keystone](#integrating_idp_and_keystone)
+* [Managing Groups](#managing_groups)
+* [Managing Mappings](#managing_mappings)
+* [Using Horizon](#using_horizon)
+* [Using the CLI](#using_the_CLI)
+* [Additional Notes](#additional_notes)
+
+## <a name="what_is_saml"></a>What is SAML Federation
+Security Assertion Markup Language (**SAML**) is a protocol that allows
+a user to use Single Sign On (**SSO**) with an Identity Provider (**IDP**)
+and a Service Provider (**SP**). When a user logs in, they give their
+credentials to the IDP. The IDP checks their credentials and a SAML
+Assertion with the user's attributes is created. That assertion is then
+passed to the SP and the SP verifies that it was sent by the IDP.
+Since the SP trusts assertions from the IDP, the user never has
+to send their credentials to the SP.
+
+Keystone will act as the SP and will authenticate users by receiving
+assertions from an IDP. In this documentation, we will use a fictitious
+SAML IDP called `Company XYZ IDP`.
+
+**Please note that any administrative commands on this page requires
+the user to have `cloud_admin` access.
+
+## <a name="integrating_idp_and_keystone"></a>Integrating the IDP and Keystone
+A Blue Box operator will need to work with a `Company XYZ IDP` integration
+engineer to setup both systems to work together.
+
+A Blue Box operator will need the following:
+* IDP Metadata URL Endpoint
+* IDP entity ID from the IDP Metadata
```

Comment:
What if there is no metadata URL? Are there specific things that we should require the IdP to provide in lieu of the metadata URL?